src/compositions/useOptionalPropertiesSchema.ts

```diff
@@ -0,0 +1,20 @@
+import { ComputedRef, MaybeRef, computed, ref } from 'vue'
+import { SchemaResponse } from '@/models'
+import { Schema } from '@/types'
+
+export type UseOptionalPropertiesSchema = {
+  schema: ComputedRef<Schema>,
+}
+
+export function useOptionalPropertiesSchema(rawSchema: MaybeRef<SchemaResponse | Schema>): UseOptionalPropertiesSchema {
+  const rawSchemaRef = ref(rawSchema)
+
+  const computedSchema = computed(() => {
+    rawSchemaRef.value.required = []
+    return rawSchemaRef.value
+  })
```

Comment:
This should probably return a new schema rather than modifying the ref. Because if a ref is passed in then it'll modify the original ref which might be unexpected.